File: HackRank-DataStructure/Array and Linked List/Compare two linked lists.cpp
bool compare_lists(SinglyLinkedListNode* head1, SinglyLinkedListNode* head2) {

    while(head1 != 0 && head2 != 0)
    {
        if(head1->data != head2->data)
            return false;
        head1=head1->next;
        head2=head2->next;
    }
    if(head1 != head2) return false;
    return true;
}

File: HackRank-DataStructure/Array and Linked List/Compare two linked lists.py
def compare_lists(llist1, llist2):
    while(llist1 != None and llist2 != None):
        if(llist1.data != llist2.data):
            return False
        llist1 = llist1.next
        llist2 = llist2.next
    if(llist1 != llist2):
        return False
    return True